src/amici.cpp

```diff
@@ -103,7 +103,15 @@ std::unique_ptr<ReturnData> runAmiciSimulation(Solver &solver, const ExpData *ed
         rdata->invalidate(model.t0());
         rdata->status = AMICI_ERROR;
         amici::warnMsgIdAndTxt("AMICI:mex:simulation","AMICI simulation failed:\n%s\nError occured in:\n%s",ex.what(),ex.getBacktrace());
-    } 
+    } catch (std::exception const& ex) {
+        model.setFixedParameters(originalFixedParameters);
+        model.setTimepoints(originalTimepoints);
+        throw ex;
+    } catch (...) {
+        model.setFixedParameters(originalFixedParameters);
+        model.setTimepoints(originalTimepoints);
+        throw std::runtime_error("Unknown internal error occured!");
+    }
     model.setFixedParameters(originalFixedParameters);
     model.setTimepoints(originalTimepoints);
     rdata->applyChainRuleFactorToSimulationResults(&model);
```

Comment:
We assume that if it is not an integration exception, there are no usable sensitivities at all? Otherwise we would have to include this line before rethrowing.

Reply:
If we throw and exception here, rdata will not be returned.

Reply:
Obviously... :man_facepalming: